src/passes/identifierManglerPass/declarationNameMangler.ts

```diff
@@ -87,46 +83,38 @@ export function checkSourceTerms(term: string, node: ASTNode) {
 }
 
 export class DeclarationNameMangler extends ASTMapper {
-  lastUsedId = 0;
-  // Feel free to increase this value. The greater the value the less probably to
-  // execute the code to fix Id's width, but less legible variable names would be.
-  initialIdWidth = 2;
-  nodesNameModified: ASTNode[] = [];
-
-  // This strategy should allow checked demangling post transpilation for a more readable result
-  createNewFunctionName(fd: FunctionDefinition, ast: AST): string {
-    return !isNameless(fd) ? `${fd.name}_${safeCanonicalHash(fd, ast)}` : fd.name;
-  }
-
-  // Return a new id formatted to achieve the minimum length
-  getFormattedId(): string {
-    return (this.lastUsedId++).toString().padStart(this.initialIdWidth, '0');
-  }
-
-  createNewTypeName(existingName: string): string {
-    return `${MANGLED_TYPE_NAME}${this.getFormattedId()}_${existingName}`;
-  }
-
-  createNewVariableName(existingName: string): string {
-    return `${MANGLED_LOCAL_VAR}${this.getFormattedId()}_${existingName}`;
+  nameCounter = 0;
+  visitSourceUnit(node: SourceUnit, ast: AST): void {
+    node.vStructs.forEach((s) => this.mangleStructDefinition(s));
+    node.vFunctions.forEach((n) => this.mangleFunctionDefinition(n, ast));
+    node.vContracts.forEach((n) => this.mangleContractDefinition(n, ast));
+    this.commonVisit(node, ast);
   }
 
   visitStructDefinition(_node: StructDefinition, _ast: AST): void {
-    // struct definitions should already have been mangled at this point
-    // by visitContractDefinition and visitSourceUnit
+    // struct definitions should have already been mangled at this point
+    // by visitSourceUnit and mangleContractDefinition
     return;
   }
```

Comment:
If this function does nothing, I think it is safe to erase it. I don't think it affects readability because by non existing you can infer structs mangling get handled somewhere else.

Reply:
Well, if you remove this function, then it will execute the code of the `visitStructDefinition` function in the base class, which does `commonVisit` on its children, so at the end, the node would be visited again and thus the mangling will be done again

Reply:
With the last changes made, it's true that this function is no longer necessary, so I removed it as suggested.